### PR TITLE
[FLINK-29372] Add suffix to all options that conflict with YAML

### PR DESCRIPTION
**Pull request by zentol**
We have a few options where the key is a prefix of other options (e.g., high-availability and high-availability.cluster-id.

Add a suffix to these options and keep the old key as deprecated.

### CODE REVIEW

**Review by pnowojski — COMMENTED** [review]
+1 I've skimmed over and didn't notice anything of concern

**Comment by austince** [comment]
Is it possible to guard against regression in the case of new config keys with some test? Maybe running through a different YAML parser?

**Comment by zentol** [comment]
> Is it possible to guard against regression in the case of new config keys with some test? Maybe running through a different YAML parser?

We could collect all keys and verify that none is a prefix of another key. That is quite simple (and how I found the offending options).

**Review by Myasuka — COMMENTED** [review]
I think we might miss some places to change:
`high-availability`: 
1. [common.sh](https://github.com/apache/flink/blob/b5cd9f34ab73fa69a3db5a09908c1aa954ed0597/flink-end-to-end-tests/test-scripts/common.sh#L201)
2. [standalone-overview](https://github.com/apache/flink/blob/master/docs/content/docs/deployment/resource-providers/standalone/overview.md#example-standalone-ha-cluster-with-2-jobmanagers)

`state.backend`:
1. [state_backend.py](https://github.com/apache/flink/blob/b5cd9f34ab73fa69a3db5a09908c1aa954ed0597/flink-python/pyflink/datastream/state_backend.py#L390)
2. [FsStateBackend.java](https://github.com/apache/flink/blob/b5cd9f34ab73fa69a3db5a09908c1aa954ed0597/flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java#L81)
3. [MemoryStateBackend.java](https://github.com/apache/flink/blob/b5cd9f34ab73fa69a3db5a09908c1aa954ed0597/flink-runtime/src/main/java/org/apache/flink/runtime/state/memory/MemoryStateBackend.java#L76)

**Comment by wanglijie95** [comment]
Following configuration options are missing：
`kubernetes.jobmanager.cpu` `kubernetes.jobmanager.cpu.limit-factor`
`kubernetes.taskmanager.cpu` `kubernetes.taskmanager.cpu.limit-factor`

**Comment by zentol** [comment]
@flinkbot run azure

**Review by Myasuka — APPROVED** [review]
Thanks for the update, LGTM.

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java [hunk 1]
```diff
@@ -269,9 +269,10 @@ public class KubernetesConfigOptions {
     @Documentation.OverrideDefault(
             "The default value depends on the actually running version. In general it looks like \"flink:<FLINK_VERSION>-scala_<SCALA_VERSION>\"")
     public static final ConfigOption<String> CONTAINER_IMAGE =
-            key("kubernetes.container.image")
+            key("kubernetes.container.image.ref")
```
Comment (zentol) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java:272]: There does not appear to be a proper term for this, so I used "ref" similar to what the log4j uses.

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java [hunk 2]
```diff
@@ -447,7 +450,7 @@ public class KubernetesConfigOptions {
                                     + Constants.MAIN_CONTAINER_NAME
                                     + "'. If not explicitly configured, config option '"
                                     + KUBERNETES_POD_TEMPLATE_FILE_KEY
-                                    + "' will be used.");
+                                    + ".default' will be used.");
 
     public static final ConfigOption<String> TASK_MANAGER_POD_TEMPLATE =
```
Comment (wangyang0918) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java:455]: We also need to update the fallback keys.

```
.withFallbackKeys(KUBERNETES_POD_TEMPLATE_FILE_KEY + ".default")

**Pull request merged**